Intermittent issue in feature_proxy.py AssertionError: not(bytearray(b'node.noumenon') == b'fc00:1:2:3:4:5:6:7')

https://drahtbot.space/temp_scratch/feature_proxy_179.tar.zstd

```
 test  2024-04-13T16:24:13.025000Z TestFramework (DEBUG): Test: outgoing CJDNS connection through node 4 for address [fc00:1:2:3:4:5:6:7]:8888 
 node4 2024-04-13T16:24:13.089818Z [http] [httpserver.cpp:306] [http_request_cb] [http] Received a POST request for / from 127.0.0.1:53136 
 node0 2024-04-13T16:24:13.090252Z [net] [net.cpp:2124] [SocketHandlerConnected] [net] socket recv error for peer=5: Connection reset by peer (104) 
 node0 2024-04-13T16:24:13.090319Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=5 
 node0 2024-04-13T16:24:13.090336Z [net] [net.cpp:2124] [SocketHandlerConnected] [net] socket recv error for peer=6: Connection reset by peer (104) 
 node0 2024-04-13T16:24:13.090347Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=6 
 node0 2024-04-13T16:24:13.090362Z [net] [net.cpp:2124] [SocketHandlerConnected] [net] socket recv error for peer=7: Connection reset by peer (104) 
 node0 2024-04-13T16:24:13.090372Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=7 
 node0 2024-04-13T16:24:13.090387Z [net] [net.cpp:2124] [SocketHandlerConnected] [net] socket recv error for peer=8: Connection reset by peer (104) 
 node0 2024-04-13T16:24:13.090397Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=8 
 node0 2024-04-13T16:24:13.090538Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=5 
 node0 2024-04-13T16:24:13.090588Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=6 
 node0 2024-04-13T16:24:13.090629Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=7 
 node0 2024-04-13T16:24:13.090670Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=8 
 node3 2024-04-13T16:24:13.091019Z [net] [net.cpp:1497] [MarkBytesSent] [net] start sending v2 handshake to peer=2 
 node3 2024-04-13T16:24:13.091068Z [net] [net.cpp:2113] [SocketHandlerConnected] [net] socket closed for peer=2 
 node3 2024-04-13T16:24:13.091080Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=2 
 node3 2024-04-13T16:24:13.091115Z [net] [net.cpp:1497] [MarkBytesSent] [net] start sending v2 handshake to peer=3 
 node3 2024-04-13T16:24:13.091155Z [net] [net.cpp:2113] [SocketHandlerConnected] [net] socket closed for peer=3 
 node3 2024-04-13T16:24:13.091173Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=3 
 node3 2024-04-13T16:24:13.091208Z [net] [net.cpp:1497] [MarkBytesSent] [net] start sending v2 handshake to peer=4 
 node3 2024-04-13T16:24:13.091226Z [net] [net.cpp:2113] [SocketHandlerConnected] [net] socket closed for peer=4 
 node3 2024-04-13T16:24:13.091245Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=4 
 node3 2024-04-13T16:24:13.091294Z [net] [net.cpp:1891] [DisconnectNodes] [net] retrying with v1 transport protocol for peer=2 
 node3 2024-04-13T16:24:13.091318Z [net] [net.cpp:1891] [DisconnectNodes] [net] retrying with v1 transport protocol for peer=3 
 node3 2024-04-13T16:24:13.091340Z [net] [net.cpp:1891] [DisconnectNodes] [net] retrying with v1 transport protocol for peer=4 
 node3 2024-04-13T16:24:13.091383Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=2 
 node3 2024-04-13T16:24:13.091447Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=3 
 node3 2024-04-13T16:24:13.091501Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=4 
 node4 2024-04-13T16:24:13.091978Z [net] [net.cpp:1497] [MarkBytesSent] [net] start sending v2 handshake to peer=2 
 node4 2024-04-13T16:24:13.092016Z [net] [net.cpp:2113] [SocketHandlerConnected] [net] socket closed for peer=2 
 node4 2024-04-13T16:24:13.092028Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=2 
 node4 2024-04-13T16:24:13.092063Z [net] [net.cpp:1497] [MarkBytesSent] [net] start sending v2 handshake to peer=3 
 node4 2024-04-13T16:24:13.092082Z [net] [net.cpp:2113] [SocketHandlerConnected] [net] socket closed for peer=3 
 node4 2024-04-13T16:24:13.092092Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=3 
 node4 2024-04-13T16:24:13.092124Z [net] [net.cpp:1497] [MarkBytesSent] [net] start sending v2 handshake to peer=4 
 node4 2024-04-13T16:24:13.092142Z [net] [net.cpp:2113] [SocketHandlerConnected] [net] socket closed for peer=4 
 node4 2024-04-13T16:24:13.092152Z [net] [net.cpp:541] [CloseSocketDisconnect] [net] disconnecting peer=4 
 node4 2024-04-13T16:24:13.092200Z [net] [net.cpp:1891] [DisconnectNodes] [net] retrying with v1 transport protocol for peer=2 
 node4 2024-04-13T16:24:13.092225Z [net] [net.cpp:1891] [DisconnectNodes] [net] retrying with v1 transport protocol for peer=3 
 node4 2024-04-13T16:24:13.092246Z [net] [net.cpp:1891] [DisconnectNodes] [net] retrying with v1 transport protocol for peer=4 
 node4 2024-04-13T16:24:13.092322Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=2 
 node4 2024-04-13T16:24:13.092386Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=3 
 node4 2024-04-13T16:24:13.092438Z [net] [net_processing.cpp:1708] [FinalizeNode] [net] Cleared nodestate for peer=4 
 node4 2024-04-13T16:24:13.092791Z [httpworker.1] [rpc/request.cpp:187] [parse] [rpc] ThreadRPCServer method=addnode user=__cookie__ 
 node4 2024-04-13T16:24:13.092861Z [httpworker.1] [net.cpp:410] [ConnectNode] [net] trying v2 connection [fc00:1:2:3:4:5:6:7]:8888 lastseen=0.0hrs 
 node4 2024-04-13T16:24:13.092899Z [httpworker.1] [net.cpp:481] [ConnectNode] [proxy] Using proxy: 127.0.0.1:13155 to connect to fc00:1:2:3:4:5:6:7:8888 
 node4 2024-04-13T16:24:13.093027Z [httpworker.1] [netbase.cpp:365] [Socks5] [net] SOCKS5 connecting fc00:1:2:3:4:5:6:7 
 node4 2024-04-13T16:24:13.093785Z [httpworker.1] [netbase.cpp:482] [Socks5] [net] SOCKS5 connected fc00:1:2:3:4:5:6:7 
 test  2024-04-13T16:24:13.094000Z TestFramework.socks5 (DEBUG): Proxy: Socks5Command(1,3,bytearray(b'fc00:1:2:3:4:5:6:7'),8888,None,None) 
 node4 2024-04-13T16:24:13.094187Z [httpworker.1] [net.cpp:3694] [CNode] [net] Added connection peer=5 
 node4 2024-04-13T16:24:13.094222Z [httpworker.1] [net.cpp:3738] [PushMessage] [net] sending version (114 bytes) peer=5 
 node4 2024-04-13T16:24:13.094253Z [httpworker.1] [net_processing.cpp:1559] [PushNodeVersion] [net] send version message: version 70016, blocks=0, txrelay=1, peer=5 
 test  2024-04-13T16:24:13.095000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
                                       self.run_test()
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/feature_proxy.py", line 252, in run_test
                                       self.node_test(self.nodes[4],
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/feature_proxy.py", line 202, in node_test
                                       assert_equal(cmd.addr, b"fc00:1:2:3:4:5:6:7")
                                     File "/ci_container_base/ci/scratch/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/util.py", line 74, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(bytearray(b'node.noumenon') == b'fc00:1:2:3:4:5:6:7')

--- a/test/functional/feature_proxy.py
+++ b/test/functional/feature_proxy.py
@@ -148,7 +148,8 @@ class ProxyTest(BitcoinTestFramework):
         rv = []
         addr = "15.61.23.23:1234"
         self.log.debug(f"Test: outgoing IPv4 connection through node {node.index} for address {addr}")
-        node.addnode(addr, "onetry")
+        # v2transport=False is used to avoid reconnections with v1 being scheduled. These could interfere with later checks.
+        node.addnode(addr, "onetry", v2transport=False)
         cmd = proxies[0].queue.get()
         assert isinstance(cmd, Socks5Command)
         # Note: bitcoind's SOCKS5 implementation only sends atyp DOMAINNAME, even if connecting directly to IPv4/IPv6
@@ -164,7 +165,7 @@ class ProxyTest(BitcoinTestFramework):
         if self.have_ipv6:
             addr = "[1233:3432:2434:2343:3234:2345:6546:4534]:5443"
             self.log.debug(f"Test: outgoing IPv6 connection through node {node.index} for address {addr}")
-            node.addnode(addr, "onetry")
+            node.addnode(addr, "onetry", v2transport=False)
             cmd = proxies[1].queue.get()
             assert isinstance(cmd, Socks5Command)
             # Note: bitcoind's SOCKS5 implementation only sends atyp DOMAINNAME, even if connecting directly to IPv4/IPv6
@@ -180,7 +181,7 @@ class ProxyTest(BitcoinTestFramework):
         if test_onion:
             addr = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion:8333"
             self.log.debug(f"Test: outgoing onion connection through node {node.index} for address {addr}")
-            node.addnode(addr, "onetry")
+            node.addnode(addr, "onetry", v2transport=False)
             cmd = proxies[2].queue.get()
             assert isinstance(cmd, Socks5Command)
             assert_equal(cmd.atyp, AddressType.DOMAINNAME)
@@ -195,7 +196,7 @@ class ProxyTest(BitcoinTestFramework):
         if test_cjdns:
             addr = "[fc00:1:2:3:4:5:6:7]:8888"
             self.log.debug(f"Test: outgoing CJDNS connection through node {node.index} for address {addr}")
-            node.addnode(addr, "onetry")
+            node.addnode(addr, "onetry", v2transport=False)
             cmd = proxies[1].queue.get()
             assert isinstance(cmd, Socks5Command)
             assert_equal(cmd.atyp, AddressType.DOMAINNAME)
@@ -209,7 +210,7 @@ class ProxyTest(BitcoinTestFramework):
 
         addr = "node.noumenon:8333"
         self.log.debug(f"Test: outgoing DNS name connection through node {node.index} for address {addr}")
-        node.addnode(addr, "onetry")
+        node.addnode(addr, "onetry", v2transport=False)
         cmd = proxies[3].queue.get()
         assert isinstance(cmd, Socks5Command)
         assert_equal(cmd.atyp, AddressType.DOMAINNAME)

